src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -425,5 +426,50 @@ internal static bool IsDeclarationExpressionType(SyntaxNode node, out Declaratio
             parent = node.Parent as DeclarationExpressionSyntax;
             return node == parent?.Type;
         }
+
+        /// <summary>
+        /// Given an initializer expression infer the name of anonymous property.
+        /// Returns default(SyntaxToken) if unsuccessful
+        /// </summary>
+        public static SyntaxToken ExtractAnonymousTypeMemberName(this ExpressionSyntax input)
```

Comment:
as a public API we likely want to give this a good name.  maybe TryGetInferredMemberName and document that it's for AnonymousTypes + Tuples?